reana_cluster/cli/cluster.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of REANA.
+# Copyright (C) 2017 CERN.
+#
+# REANA is free software; you can redistribute it and/or modify it under the
+# terms of the GNU General Public License as published by the Free Software
+# Foundation; either version 2 of the License, or (at your option) any later
+# version.
+#
+# REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+# A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along with
+# REANA; if not, write to the Free Software Foundation, Inc., 59 Temple Place,
+# Suite 330, Boston, MA 02111-1307, USA.
+#
+# In applying this license, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization or
+# submit itself to any jurisdiction.
+"""REANA cluster administration (init, restart, etc.) commands."""
+
+
+import logging
+
+import click
+import os
+
+import errno
+import yaml
+
+from ..config import generated_cluster_conf_default_path
+
+from reana_cluster import KubernetesBackend
+
+from ..utils import load_spec_file
+
+
+@click.command()
+@click.option('--remove-persistent-storage', is_flag=True,
+              help='If set, also persistent storage '
+                   'inside the cluster is deleted.')
+@click.pass_context
+def down(ctx, remove_persistent_storage):
+    """Bring REANA cluster down, i.e. shutdown all components."""
+
+    try:
+        cluster_spec = load_spec_file(click.format_filename(
```

Comment:
can we bring this code to the `cli` group command in `reana_cluster/cli/__init__.py`? and provide the `cluster_spec` object inside `ctx`

Reply:
Maybe. I was even thinking to write a decorator for it. 

If we bring this to __init__ we make implicit assumption that all of the commands require the specification file to be parsed. This might be the case, but I have to check. On the other hand `-f / --file` is already there so valid point :)

Currently `reana-cluster` is connecting to Kubernetes based on `~/.kube/config` and not according to what is specified in `cluster-specification.yaml`, and as such we don't necessarily need cluster specification file parsing for `get <component_name>` -functionality. Although `init()` of `KubernetesBackend` does require.

> Currently `reana-cluster` is connecting to Kubernetes based on `~/.kube/config` and not according to what is specified in `cluster-specification.yaml`

It might be too much to define Kubernetes config here as we need certs, etc. authorization stuff in there, but this is kind of the question behind this whole discussion; what should be included in cluster / backend -part of reana cluster specification?

Reply:
> what should be included in cluster / backend -part of reana cluster specification

Maybe even just the kubectl location?
I have a vague memory that there was an environment variable available that points to the currently active kubeconfig-file for `kubectl`. Maybe we can use that as a default.

Of course this would require that user of reana-cluster has `kubectl` available, but maybe this is fair assumption since `reana-cluster` is more of an admin / developer tool.